pybind/mgr/restful: improve cert handling; work with vstart

liewegas opened the pull request.

tchaikov commented:
retest this please. see #15406

b-ranto commented on src/pybind/mgr/restful/module.py:
```diff
@@ -244,18 +245,43 @@ def _serve(self):
+        cert_fname = None
+        pkey_fname = None
+
+        cert = self.get_config("cert")
```
This should be get_config_json, not get_config. get_config returns the cluster config as in /config/cluster, not the config-keys. (the same should go for other get_config calls).

liewegas replied:
i'm not sure what you mean. see the config-key sets in vstart.sh; this is pulling those same values (and working).  same thing that dashboard is doing...

b-ranto replied:
Ah, sorry, I have misread that. I thought it was self.get('config') which is a completely different thing. I suppose this should work. The only question is whether we want to do the json serialization or not but I guess for the keys, it does make sense not to serialize.

b-ranto commented on src/pybind/mgr/restful/module.py:
```diff
@@ -244,18 +245,43 @@ def _serve(self):
+                      server_addr, server_port)
+
+        cert_fname = None
+        pkey_fname = None
```
There is no need to initialize variables like this in python, they will be overwritten (re-initialized) by next assignment anyway (that holds even for if/else like you use later in the code). The same goes for cert_tmp and pkey_tmp.

liewegas replied:
ah, fixed

liewegas commented:
@b-ranto can you take another quick look at the test change too?

b-ranto commented:
lgtm as well

liewegas commented:
retest this please

liewegas commented:
@b-ranto @jcsp: several changes

- changed commands to have 'restful' prefix
- added get_mgr_id
- added get_config_prefix
- get_localized_config helper in restful
- used 'crt' and 'key' names instead of 'cert' and 'pkey' (to match file extensions)

liewegas commented:
retest this please

liewegas commented:
http://pulpito.ceph.com/sage-2017-06-05_22:19:51-rados-wip-sage-testing-distro-basic-smithi/

liewegas commented:
http://pulpito.ceph.com/sage-2017-06-05_22:19:51-rados-wip-sage-testing-distro-basic-smithi/ looks ok, but make check needs retset

liewegas commented:
retest this please

liewegas commented:
retest this please

liewegas commented:
retest this please
